src/snowflake/connector/telemetry.py

```diff
@@ -36,16 +37,36 @@ class TelemetryField(Enum):
     ARROW_FETCH_ALL = "client_fetch_arrow_all"
     ARROW_FETCH_BATCHES = "client_fetch_arrow_batches"
     # Keys for telemetry data sent through either in-band or out-of-band telemetry
-    KEY_TYPE = "type"
-    KEY_SOURCE = "source"
+    KEY_TYPE = "Type"
+    KEY_SOURCE = "Source"
     KEY_SFQID = "QueryID"
     KEY_SQLSTATE = "SQLState"
     KEY_DRIVER_TYPE = "DriverType"
     KEY_DRIVER_VERSION = "DriverVersion"
-    KEY_REASON = "reason"
-    KEY_ERROR_NUMBER = "ErrorNumber"
-    KEY_STACKTRACE = "Stacktrace"
+    KEY_REASON = "Reason"
+    KEY_VALUE = "Value"
+    KEY_REQUEST = "Request"
+    KEY_ERROR_CODE = "ErrorCode"
+    KEY_RESPONSE = "Response"
+    KEY_RESPONSE_STATUS_LINE = "ResponseStatusLine"
+    KEY_RESPONSE_STATUS_CODE = "ResponseStatusCode"
+    KEY_RETRY_TIMEOUT = "RetryTimeout"
+    KEY_RETRY_COUNT = "RetryCount"
     KEY_EXCEPTION = "Exception"
+    KEY_EXCEPTION_MESSAGE = "ExceptionMessage"
+    KEY_EXCEPTION_STACK_TRACE = "ExceptionStackTrace"
+    KEY_EVENT_TYPE = "EventType"
+    KEY_EVENT_SUB_TYPE = "EventSubType"
+    KEY_SFC_PEER_HOST = "SfcPeerHost"
+    KEY_CERT_ID = "CertId"
+    KEY_OCSP_REQUEST_BASE64 = "OcspRequestBase64"
+    KEY_OCSP_RESPONDER_URL = "OcspResponderURL"
+    KEY_INSECURE_MODE = "InsecureMode"
+    KEY_FAIL_OPEN = "FailOpen"
+    KEY_CACHE_ENABLED = "CacheEnabled"
+    KEY_CACHE_HIT = "CacheHit"
+    KEY_TELEMETRY_SERVER_DEPLOYMENT = "TelemetryServerDeployment"
+    KEY_CONNECTION_STRING = "ConnectionString"
```

Comment:
These should also all be snake_case. In Snowpark Scala they also use snake_case. See here: https://github.com/snowflakedb/thundersnow/blob/master/src/main/scala/com/snowflake/snowpark/internal/Telemetry.scala

Only OOB keys should be camelCase

Reply:
fixed